src/coreclr/src/System.Private.CoreLib/src/System/Type.CoreCLR.cs

```diff
@@ -20,27 +21,31 @@ public bool IsInterface
             }
         }
 
+        [RequiresUnreferencedCode("The type might be removed")]
```

Comment:
Yes - the intrinsic handling takes precedence over any attributes - in fact the attribute will do "nothing" for callers with the current linker since the intrinsic handling will always kick in (and either work or report a more specific warning). The attribute is mostly added for:
* Potential tools which won't have intrinsic handling of this
* The body of the method - as the attribute also suppresses all warnings coming from the method's body